fastlane/fastlane
# [fastlane/gym] add rubocop rule for FastlaneCore::ConfigItem `is_string` (applied to gym)

### Checklist

### Motivation and Context


> `FastlaneCore::ConfigItem` has deprecated parameter called is_string. To prevent setting a value of this parameter, introducing rubocop rule would be very helpful.



closes https://github.com/fastlane/fastlane/issues/15179

### Description



Added a new rubocop rule called based on [PR](https://github.com/fastlane/fastlane/pull/13816) from @joshdholtz ❤️ 

To proof that a rule is working as expected, it has been enabled for `gym` tool 👇 

```
Lint/IsStringUsage:
  Include:
    - 'gym/**/*'
```

We will introduce more PRs where we will get rid of `is_string` param tool by tool. 

### Testing Steps



Run `bundle exec rubocop` to check if there are no offenses detected (psst, you can also check a CI job) 🎉 

🙇 🙇
